unic/ucd/common/tests/consistency_tests.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+#[macro_use]
+extern crate unic_char_range;
+
+extern crate unic_ucd_category;
+extern crate unic_ucd_common;
+
+
+use unic_ucd_category::GeneralCategory;
+use unic_ucd_common::{is_alphabetic, is_alphanumeric, is_control, is_numeric, is_white_space};
+
+
+macro_rules! assert_char {
+    ($ch:expr, $x:expr => $y:expr) => (
+        assert!(
+            !$x || $y,
```

Comment:
These are logical equivalents, but given the below textual assertion, I think this would read better as `!($x && !$y)`. I _think_ that is closer to the actual assertion here: that if we have `$x` then `$y` must be true. Maybe, it's better written explicitly: `if $x { assert!( $y, ... ) }`

But this is a stylistic choice. I'll leave it up to you.

Reply:
Thanks, @CAD97, for the comments!

The reason the precondition is inside the assertion macro to be able to write out a meaningful error on failure. First I was only passing in `$y` to the macro, then realized that the text needs to know about the precondition, and it doesn't make sense to repeat it on each macro call-site.

About changing the logical form, IMHO the current form is more common and easier to read. The double-negation on the other form makes it harder to follow. (I think it's similar to the case we had last week, where we went for the simpler logical formula.)

There's only one thing that I want to change here, though, which is `=>`, which is a valid operator for the expressions in this context, and therefore, I want to update it to `==>`, `->` or `-->`, which cannot be part of the expressions.